docs/blog/_posts/2020-12-18-scala3-m3.md

```diff
@@ -0,0 +1,203 @@
+Previously, context function closures were written as follows:
+
+```scala
+(using s: Show[String]) => s.show("foobar")
+```
+
+Now, we have reverted to the following syntax:
+```scala
+(s: Show[String]) ?=> s.show("foobar")
+```
+
+# `Matchable` trait – a new top-level type
+The motivation for this change is the fact that it is possible to access the underlying type referenced by an opaque type using pattern matching:
+
+```scala
+val imm = IArray(1,2,3) // supposedly immutable...
+imm match
+   case a: Array[Int] => a(0) = 0  // but that's shown to be lie
+```
+
+To address this change, a new trait, `Matchable`, is introduced at the top of the type hierarchy:
+
+```scala
+abstract class Any:
+  def asInstanceOf
+  def ==
+  def !=
+  def ##
+  def equals
+  def hashCode
+  def toString
+
+trait Matchable extends Any:
+  def isInstanceOf
+  def getClass
+
+class AnyVal extends Any, Matchable
+
+class Object extends Any, Matchable
+```
+
+With the addition of this type, we now make the compiler emit warnings when trying to do a trick like the one discussed above:
+
+```scala
+-- Warning: i7314.scala:6:12 ---------------------------------------------------
+6 |    case a: Array[Int] =>
+  |            ^^^^^^^^^^
+  |            pattern selector should be an instance of Matchable,
+  |            but it has unmatchable type opaques.IArray[Int] instead
+```
+
+The change involved an extensive discussion which you can read in the [PR #10670](https://github.com/lampepfl/dotty/pull/10670).
+
+# Tooling improvements
+As the work on Scala 3 gets closer to the stable release, the focus increasingly shifts on the tooling available to get started with Scala 3.
+
+As part of the tooling effort, Scala 3 documentation tool is rapidly improved. [PR #10522](https://github.com/lampepfl/dotty/pull/10522) proves that the doctool can generate documentation for the community build projects.
+
+[PR #10491](https://github.com/lampepfl/dotty/pull/10491) introduced the scripting capability to Scala 3. Consider the following source named e.g. `Main.scala`:
+
+```scala
+@main def Test(name: String): Unit =
+  println(s"Hello ${name}!")
+```
+
+If you have Scala 3 binaries on your path (which you can get by following the steps at the [Dotty website](https://dotty.epfl.ch/)'s section "Try DOtty"), you can run the `scala Main.scala World` command. This command will compile the source in question to a temporary directory and run the discovered main method with the argument `World`.
+
+Note the difference from the Scala 2 scripting implementation. In Scala 2, we do not require the users to have a `main` method in their scripts due to it being too cumbersome to write. In Scala 3, thanks to the top-level definitions and the `@main` annotations, `main` methods are one-liners and hence are more suited for scripts.
+
+The documentation for this feature is available [here](https://dotty.epfl.ch/docs/usage/getting-started.html#scala-3-for-scripting).
+
+# Metaprogramming changes
```

Comment:
/cc @nicolasstucki

Reply:
@anatoliykmetyuk the markdown format is wrong in this section `(..)[..]` vs `[..](..)`